### ISSUE TITLE
Warn Windows 7 users when Powershell is the wrong version

### ISSUE THREAD

**Issue by russlamb** [electron/electron-packager#1029]
### Preflight Checklist

* [X ] I have read the [contribution documentation](https://github.com/electron-userland/electron-packager/blob/master/CONTRIBUTING.md) for this project.
* [X ] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [X ] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description
I would like an error message to be displayed when the wrong version of Powershell is detected.  

Currently, electron-packager requires a powershell version of 3 or later.  Yet all Windows 7 machines only have Powershell version 2 installed by default.  If using the wrong version of Powershell, the electron-packager simply hangs at the extracting step without displaying any error or warning.  

As such, the only way to find out the cause of the problem is to come to this forum.  

### Proposed Solution
I propose that electron packager check the Powershell version if on windows platform and throw an error if it detects the wrong version.  

The powershell version can be detected by running the following command from the system command line:

powershell -Command "$PSVersionTable.PSVersion"

### Alternatives Considered
Currently the README lists Powershell 3 as one of its dependencies, but Powershell version is not something most Windows users are aware of.  

### Additional Information

Judging by the number of related issues raised on this board, it seems this is a fairly common issue for people on Windows 7 machines.  Here are some examples
[Example 1](https://github.com/electron-userland/electron-packager/issues/1022)
[Example 2](https://github.com/electron-userland/electron-packager/issues/1018)
[Example 3](https://github.com/electron-userland/electron-packager/issues/1025)

I had the same issue myself prior to coming to these boards.  While I appreciate the effort the dev and support contributors have made to resolve and document these problems it would make electron packager easier for new developers to use if an error were thrown instead of just hanging.  

While I understand that the README file lists dependencies, it would make electon-packager an all-around better experience if it throws errors when the Powershell dependency is missing rather than silently hanging.  Powershell is not something many Windows users use and therefore may not be obvious what the problem is.

**Comment by malept** [comment]
I would be happy to review a pull request for this. I have a WIP branch to warn Windows 7 users. The main problem for me right now is writing tests for this.

Also, I don't have a Windows 7 instance to work with at the moment, so I'm kind of flying blind.

**Comment by malept** [comment]
> I have a WIP branch to warn Windows 7 users.

See https://github.com/electron-userland/electron-packager/pull/1030 for a first draft.

**Comment by russlamb** [comment]
awesome, thanks for turning this around!

I would test for you but i just upgraded my powershell.  If I can figure
out how to downgrade I will follow up.

What's the best way to install the changed version?

**Comment by malept** [comment]
> I would test for you but i just upgraded my powershell. 

The first iteration doesn't check the Powershell version. It only warns Windows 7 users to make sure they have the right .NET Framework and Powershell versions installed. However, it also provides a better code location for a contributor with access to Windows 7 to work on another iteration that actually tests for .NET Framework and Powershell versions.

> What's the best way to install the changed version?

In package.json `devDependencies`:

```json
{
  "devDependencies": {
    "electron-packager": "electron-userland/electron-packager#warn-windows-7"
  }
}
```

**Comment by russlamb** [comment]
How do I make sure electron-packager is using the local version and not the
global version?

How do I uninstall global package?

**Comment by russlamb** [comment]
nevermind, i uninstalled the global package and reinstalled using the
#warn-windows-7 URL.  I'm getting seeing the warning text you added.
thanks again!

**Comment by ziggyzulu** [comment]
Hi guys, using Windows 7 here - and along with all the others I too am hanging on this line:

"Packaging app for platform win32 x64 using electron v5.0.2
WARNING: Make sure that .NET Framework 4.5 or later and Powershell 3 or later are installed, otherwise extracting the Electron zip file will hang."

I'm pretty sure I have the right version of Powershell. When I run the command "powershell -Command "$PSVersionTable.PSVersion" I get:

Major: 3
Minor: 0
Build: -1
Revision -1

I also downloaded the .NET Developer pack, and I believe I also downloaded the Runtime version for good measure.

I re installed electron-packager in hopes that would solve it - but it didn't. 

I'm at a bit of a dead end, but I'm sure the solution is something simple. Has anyone any suggestions?
